core-domain/src/main/java/org/uniprot/core/proteome/impl/ProteomeStatisticsImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.uniprot.core.proteome.impl;
+
+import java.util.Objects;
+
+import org.uniprot.core.proteome.ProteomeStatistics;
+
+public class ProteomeStatisticsImpl implements ProteomeStatistics {
+    private final long reviewedProteinCount;
```

Comment:
you can extends StatisticsImpl for code-reuse. After that you can remove getReviewedProteinCount and getUnreviewedProteinCount from this impl class.

Reply:
Hi @supun-ebi 
Forgot to comment. Please add null check in the constructor for mandatory fields and throw exception if they are null.